mongoengine/queryset/base.py

```diff
@@ -1711,17 +1765,19 @@ def _get_order_by(self, keys):
             direction = pymongo.ASCENDING
             if key[0] == '-':
                 direction = pymongo.DESCENDING
+
             if key[0] in ('-', '+'):
                 key = key[1:]
+
             key = key.replace('__', '.')
             try:
                 key = self._document._translate_field_name(key)
             except Exception:
+                # TODO this exception should be more specific
                 pass
+
             key_list.append((key, direction))
 
-        if self._cursor_obj and key_list:
-            self._cursor_obj.sort(key_list)
```

Comment:
Glad you asked :) Manipulating the cursor within this method makes for a poor separation of concerns. I would not expect to change the state of the cursor just by trying to convert MongoEngine-like sort keys to PyMongo-like sort tuples. That job is better left to `BaseQuerySet.order_by`, which is where I moved it (https://github.com/MongoEngine/mongoengine/pull/1476/files#diff-05c70acbd0634d6d05e4a6e3a9b7d66bR1007)